src/svs/inacademia_frontend.py

```
@@ -177,6 +183,28 @@ def handle_authn_request(self, context):
 
         return self.auth_req_callback_func(context, internal_request)
 
+    def _validate_authn_request(self, request):
+        claims = request.get('claims')
+        if claims:
```

While the method is correct, I do not think that this is right. We should not specialize the main purpose of this method. We should branch out to handle the side-cases. The main business of this method is to validate a value. If there is no value we stop early; it is not the other way around.

This hints to me that we're doing something else wrong, because we should not need to return early. We are passing in the `request` but we're only interested in its `claims` member. We should be passing in just that.

---

Additionally, this method is called `_validate_[...]`. It should either return or raise an exception. If it was called `_is_valid_[...]` it should either return `True` or `False`. But, this method is doing neither; it is doing its own thing - namely, it is returning an error, which is then returned further up by the caller.

I think this is wrong. It should raise the error with the context that it has. The caller should catch the error and handle it as it needs.

---

Think these two points, I think this should be restructured.

---

Thanks, @c00kiemon5ter for your thoughtful review. I have made the suggested changes but I must mention, that initially, I implemented this method with the viewpoint that this method `_validate_authn_request` will be expanded to validate other request constraints as well which I would be working on in the coming days. For example: if client_id or redirect_uri is invalid we need to show InAcademia error page instead of redirecting to invalid RP or invalid client, also if client_id or redirect_uri do not match with the `cdb.json` data, in that case as well we need to redirect to InAcademia error page. (Error 1 - 4 ) 

But anyway, I think you are right, for now, let's concentrate on the claims.